@kpedro88 I changed lines 249 and 265.
I had also tried the snippet you wrote in this form
https://github.com/cms-sw/cmssw/pull/18846/commits/b5b5cd849b918ce41528e68105ab6b4a04887b57#diff-bb2124e47b0d956b141c59f69f2496d4R221
but if I try to compile I get the error below [*] and this is where I had stopped,
let's see if you have a way out

thanks a lot
luca

[*]
```
>> Compiling  /afs/cern.ch/work/p/perrozzi/private/git/Hbb/validations/LHERunInfoProduct/CMSSW_9_2_0_patch1/src/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc
/afs/cern.ch/work/p/perrozzi/private/git/Hbb/validations/LHERunInfoProduct/CMSSW_9_2_0_patch1/src/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc: In static member function 'static bool LHERunInfoProduct::isTagComparedInMerge(const string&)':
/afs/cern.ch/work/p/perrozzi/private/git/Hbb/validations/LHERunInfoProduct/CMSSW_9_2_0_patch1/src/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc:222:52: error: cannot call member function 'bool LHERunInfoProduct::find_if_checklist(std::string, std::vector<std::basic_string<char> >)' without object
         return !find_if_checklist(tag,tag_checklist);
                                                    ^
config/SCRAM/GMake/Makefile.rules:2017: recipe for target 'tmp/slc6_amd64_gcc530/src/SimDataFormats/GeneratorProducts/src/SimDataFormatsGeneratorProducts/LHERunInfoProduct.o' failed
gmake: *** [tmp/slc6_amd64_gcc530/src/SimDataFormats/GeneratorProducts/src/SimDataFormatsGeneratorProducts/LHERunInfoProduct.o] Error 1
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2

```

--- a/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc
+++ b/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc
@@ -206,7 +206,6 @@ bool HeaderLess::operator() (const LHERunInfoProduct::Header &a,
 }
 
 static std::vector<std::string> checklist{"iseed","Random",".log",".dat",".lhe"};
-static std::vector<std::string> tag_checklist{"","Alpgen","MGGridCard","MGRunCard","mgruncard","MadSpin","madspin"};
 static std::vector<std::string> tag_comparison_checklist{"","MGRunCard","mgruncard"};
 
 bool LHERunInfoProduct::find_if_checklist(const std::string x, std::vector<std::string> checklist) {
@@ -217,10 +216,6 @@ bool LHERunInfoProduct::find_if_checklist(const std::string x, std::vector<std::
 bool LHERunInfoProduct::isTagComparedInMerge(const std::string& tag) {
         return !(tag == "" || tag.find("Alpgen") == 0 || tag == "MGGridCard" || tag=="MGRunCard" || tag == "mgruncard" || tag=="MadSpin" || tag=="madspin");
 }
-
-// bool LHERunInfoProduct::isTagComparedInMerge(const std::string& tag) {
-        // return !find_if_checklist(tag,tag_checklist);
-// }
 
 bool LHERunInfoProduct::mergeProduct(const LHERunInfoProduct &other)
 {
@@ -287,9 +282,7 @@ bool LHERunInfoProduct::mergeProduct(const LHERunInfoProduct &other)
         if(header_compatible) continue;
       }
       
-			if( !find_if_checklist(header->tag(),tag_checklist)
-        // aka isTagComparedInMerge
-        ){ 
+			if( !isTagComparedInMerge(header->tag()) ){ 
 				failed = true;
 			} else {
 				addHeader(*header);	

--- a/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc
+++ b/SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc
@@ -282,7 +282,7 @@ bool LHERunInfoProduct::mergeProduct(const LHERunInfoProduct &other)
         if(header_compatible) continue;
       }
       
-			if( !isTagComparedInMerge(header->tag()) ){ 
+			if(isTagComparedInMerge(header->tag())){ 
 				failed = true;
 			} else {
 				addHeader(*header);	

